packages/payments-plugin/src/mollie/mollie.helpers.ts

```diff
@@ -79,6 +79,13 @@ export function toAmount(value: number, orderCurrency: string): Amount {
     };
 }
 
+/**
+ * Return to number of cents
+ */
+export function amountToCents(amount: Amount): number {
+    return Math.round(Number(amount.value) * 100); // Round because floating point errors after Number() conversion
```

Comment:
I would suggest to use a dedicate library for any currency conversions/math to avoid rounding issues in general:
<img width="715" alt="image" src="https://user-images.githubusercontent.com/79153163/226943327-ea4d5f62-632d-4834-a8f0-85ef9aa540f7.png">

Reply:
@martijnvdbrug Interesting - I was wondering about this and I see from the [Mollie SDK](https://github.com/mollie/mollie-api-node) that `amount.value` is a string like `'10.00'`. Checked some values and indeed you do get rounding issues!
```ts
Number('10.21') * 100
// => 1021.0000000000001
```

In general @StampixSMO I agree we need to be cautious with math on currency - that's why Vendure only stores integers for currency values. In this case, I think the handling is fine because we're dealing here with an artifact of "floating-point numbers in binary" rather than an actual rounding error caused by multiplying by a non-integer, so the error is only ever so tiny as to be ignored.

Reply:
Mollie returns a string on purpose, it avoids FP errors due to frameworks along the way. It's meant to be used with a professional library like currency.js for math and parsing.

Thanks for confirming the Vendure strategy. With integers, you implicitly mean that you store the least significant currency unit, right? E.g. dollarcents, eurocents, ... instead of dollars and euros.

Reply:
Oke, the library is tiny, so I'll add it :-)

Reply:
@StampixSMO yes, we store the minor unit.